src/js/views/patients/patient/sidebar/sidebar_views.js

```diff
@@ -83,19 +85,64 @@ const GroupsView = View.extend({
   },
 });
 
+const EngagementStatusView = View.extend({
+  ui: {
+    loading: '.js-loading',
+  },
+  modelEvents: {
+    'change:engagement': 'render',
+  },
+  getTemplate() {
+    if (!this.model.getEngagement().status) {
+      return hbs`<div class="js-loading">{{ @intl.patients.patient.sidebar.sidebarViews.engagementStatusView.loading }}</div>`;
+    }
+
+    return hbs`
+      {{#if statusUnavailable}}
+        <div class="patient-sidebar__no-engagement">{{ @intl.patients.patient.sidebar.sidebarViews.engagementStatusView.notAvailable }}</div>
+      {{else}}
+        <span class="engagement-status__icon {{ status }} u-margin--r-4">{{fas "circle"}}</span>{{formatMessage (intlGet "patients.patient.sidebar.sidebarViews.engagementStatusView.status") status=status}}
+      {{/if}}
+    `;
+  },
+  templateContext() {
```

Comment:
so if we move the logic into the app the view can just use `{{ engagement.status }}`